src/utils.ts

```diff
@@ -66,12 +66,24 @@ export const ensureArray = function <T> (maybeArray: T | T[]) {
   return maybeArray
 }
 
+const isSigned = (message: PubSubRPCMessage): boolean => {
+  if ((message.sequenceNumber == null) || (message.from == null) || (message.signature == null)) {
+    return false
+  }
+  // if a public key is present in the `from` field, the message should be signed
```

Comment:
Is this true?  A peer with an Ed25519/secp256k1 key would never be able to send an unsigned message in this case.